packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -249,7 +249,7 @@ export function shouldSetTextContent(type: string, props: Props): boolean {
     typeof props.children === 'number' ||
     (typeof props.dangerouslySetInnerHTML === 'object' &&
       props.dangerouslySetInnerHTML !== null &&
-      typeof props.dangerouslySetInnerHTML.__html === 'string')
+      typeof props.dangerouslySetInnerHTML.__html != null)
```

Comment:
I don't think you intended to keep the `typeof` here?

Reply:
No. Also there are other reasons why this doesn't work. Will look again.